cmd/wallet/address.go

```diff
@@ -76,12 +77,33 @@ func buildNewAddressCmd(parentCmd *cobra.Command) {
 	}
 	parentCmd.AddCommand(newAddressCmd)
 
+	addressType := newAddressCmd.Flags().String("address_type",
+		w.AddressTypeBLSAccount.String(), "Address type to create")
+
 	newAddressCmd.Run = func(_ *cobra.Command, _ []string) {
+		var addr string
+		var err error
+
 		label := cmd.PromptInput("Label")
 		wallet, err := openWallet()
 		cmd.FatalErrorCheck(err)
 
-		addr, err := wallet.NewBLSAccountAddress(label)
+		isForBLSAccount := *addressType == w.AddressTypeBLSAccount.String()
+		isForValidatorAccount := *addressType == w.AddressTypeValidator.String()
+
+		if !isForBLSAccount || !isForValidatorAccount {
```

Comment:
It is complex. You can make it simple like:

if addrType == "fo" { ... }
else if addrType = "bar" { .. }
else { error }

Feel free to use switch/case